src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -506,14 +508,17 @@ public bool AddToBalanceAndCreateIfNotExists(Address address, in UInt256 balance
             }
         }
 
+        public Task WaitForCodeCommit() => _codeFlushTask;
+
         public void Commit(IReleaseSpec releaseSpec, bool commitRoots, bool isGenesis)
             => Commit(releaseSpec, NullStateTracer.Instance, commitRoots, isGenesis);
 
         public void Commit(IReleaseSpec releaseSpec, IWorldStateTracer stateTracer, bool commitRoots, bool isGenesis)
         {
-            Task codeFlushTask = !commitRoots || _codeBatch is null || _codeBatch.Count == 0
-                ? Task.CompletedTask
-                : CommitCodeAsync();
+            if (commitRoots && _codeBatch?.Count > 0)
```

Comment:
Quick Question: Can it cause consensus issues if a few commits are made before the previous async write has finished?

We're doing multiple commits during block processing, and we might encounter a situation where the cache hasn't been fully cleared or the previous commit hasn't completed, yet we begin processing a new block.